promdump/promdump.go

```diff
@@ -651,9 +770,52 @@ func main() {
 		}
 	}
 	if *metric != "" {
-		err = exportMetric(ctx, promApi, *metric, beginTS, endTS, *periodDur, *batchDur, *batchesPerFile, *out)
+		customMetricCount, err = exportMetric(ctx, promApi, *metric, beginTS, endTS, *periodDur, *batchDur, *batchesPerFile, *out)
 		if err != nil {
 			log.Fatalln("exportMetric:", err)
 		}
 	}
+
+	if *enableTar {
+		var filename string
+		filename = *tarFilename
+		if *tarFilename == "" {
+			// If filename is not provided, generate a default tar filename
+			filename = generateDefaultTarFilename()
+		} else {
+			// Check if filename already exists
+			if _, err := os.Stat(filename); err == nil {
+				log.Fatalf("Error: %s already exists. Please choose a different filename.\n", filename)
+				return
+			}
+		}
+
+		out, err := os.Create(filename)
+		if err != nil {
+			log.Printf("Error writing archive: %v (File: %s)\n", err, filename)
```

Comment:
Use %v for both `err` and `filename`.

Reply:
- Remove the `filename`--> use *tarFilename
- Change %s to %v